docs/metrics.md

```diff
@@ -0,0 +1,561 @@
+```rust
+use opentelemetry::global;
+use opentelemetry::InstrumentationScope;
+use opentelemetry::KeyValue;
+
+let scope = InstrumentationScope::builder("my_company.my_product.my_library")
+        .with_version("0.17")
+        .with_schema_url("https://opentelemetry.io/schema/1.2.0")
+        .with_attributes([(KeyValue::new("key", "value"))])
+        .build();
+
+// creating Meter with InstrumentationScope, comprising of
+// name, schema and attributes.
+let meter = global::meter_with_scope(scope);
+
+// creating Meter with just name
+let meter = global::meter("my_company.my_product.my_library");
+```
+
+### Instruments
+
+:heavy_check_mark: You should understand and pick the right instrument type.
+
+  > [!NOTE] Picking the right instrument type for your use case is crucial to
+  > ensure the correct semantics and performance. Check the [Instrument
+    Selection](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/supplementary-guidelines.md#instrument-selection)
+    section from the supplementary guidelines for more information.
+
+| OpenTelemetry Specification | OTel Rust Instrument Type |
+| --------------------------- | -------------------- |
+| [Asynchronous Counter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-counter) | [`ObservableCounter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.ObservableCounter.html) |
+| [Asynchronous Gauge](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-gauge) | [`ObservableGauge`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.ObservableGauge.html) |
+| [Asynchronous UpDownCounter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-updowncounter) | [`ObservableUpDownCounter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.ObservableUpDownCounter.html) |
+| [Counter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#counter) | [`Counter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.Counter.html) |
+| [Gauge](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#gauge) | [`Gauge`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.Gauge.html) |
+| [Histogram](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#histogram) | [`Histogram`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.Histogram.html) |
+| [UpDownCounter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#updowncounter) | [`UpDownCounter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.UpDownCounter.html) |
+
+:stop_sign: You should avoid creating instruments (e.g. `Counter`) too
```

Comment:
Similar to https://github.com/open-telemetry/opentelemetry-rust/pull/2946#discussion_r2067407462